### ISSUE TITLE
Incorrect Versions of Jakarta Inject On Class Path

### ISSUE THREAD

**Issue by JKomoroski** [xvik/dropwizard-guicey#317]
After upgrading from version 5.8.0 to [version 5.9.0](https://github.com/xvik/dropwizard-guicey/tree/5.9.0) of dropwizard guicey, I am getting the following error:
```
java.lang.NoClassDefFoundError: jakarta/inject/Provider
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:862)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:760)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:681)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:639)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        at com.google.inject.Scopes.<clinit>(Scopes.java:37)
        at com.google.inject.internal.InjectorShell$RootModule.configure(InjectorShell.java:340)
        at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:426)
        at com.google.inject.spi.Elements.getElements(Elements.java:113)
        at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:160)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107)
        at com.google.inject.Guice.createInjector(Guice.java:87)
        at com.google.inject.Guice.createInjector(Guice.java:69)
        at com.google.inject.Guice.createInjector(Guice.java:59)
```

I rely on the guicey-bom for version managment of guice, dropwizard, javax inject, and jakarta inject. What I found is that Guice is expecting jakarta inject version `2.x`, but dropwizard version `2.1.6` expects jakarta inject `1.x`.
Links:
- https://github.com/google/guice/blob/f095cad8aa7b59abafacc4d4461ddfc77be83667/pom.xml#L142
- https://repo1.maven.org/maven2/ru/vyarus/guicey/guicey-bom/5.9.0/guicey-bom-5.9.0.pom (shows dropwizard 2.1.6)
- https://github.com/dropwizard/dropwizard/blob/50587fa6c899beaad547b4f7306157beff271375/dropwizard-dependencies/pom.xml#L42

I believe this will need to be explicitly managed in the Guicey bom in the 5.9.x + versions to work correctly for projects which defer to the 
Guicey-Bom for version management.

**Comment by xvik** [comment]
There were actually two problems: guice depends on both `jakarta.inject` and `javax.inject` (because guice 6 supports both). But, I excluded `javax.inject` entirely.

The problem with BOM fixed by ["manuallty" adding required jakarta.inject](https://github.com/xvik/dropwizard-guicey/commit/bad7fb15705bca98369e657ec3c848926788eb86) version BEFORE  dropwizard and guice BOMs import. 

I also added two maven projects and samples: with [direct dependency](https://github.com/xvik/dropwizard-guicey/tree/dw-2.1/examples/maven-simple) and [BOM](https://github.com/xvik/dropwizard-guicey/tree/dw-2.1/examples/maven-bom). They both called now as part of CI, so next time I should quickly detect regressions.

I will backport changes into 3.x. 

Dropwizard finally published oficcially the latest releases so nothing holds guicey release anymore and it would be released in a few days (as soon as I finish with other minor things).
